src/app/clusters/reporting/reporting.c

```diff
@@ -0,0 +1,1024 @@
+/**
```

Comment:
"reporting" isn't really a "cluster", right?  Should we just have `src/app/reporting` for this stuff instead?  Or some other dir under `src/app` for cluster-spanning "global" stuff like this that isn't a good fit for `util`?

Reply:
agreed makes sense to move, will move